java/vector/src/main/java/org/apache/arrow/vector/NullVector.java

```diff
@@ -155,9 +155,16 @@ public boolean allocateNewSafe() {
   @Override
   public void reAlloc() {}
 
+  /*
+   * IMPORTANT NOTE
+   * It's essential that NullVector (and ZeroVector) do not require BufferAllocator for any data storage.
+   * However, some methods of the parent interface may require passing in a BufferAllocator, even if null.
+   *
+   * @return null
+   */
   @Override
   public BufferAllocator getAllocator() {
-    throw new UnsupportedOperationException("Tried to get allocator from NullVector");
+    return null;
```

Comment:
I haven't added a test to check this returns `null` because this seems to me an implementation detail. Please let me know if you think otherwise.